torchgeo/datasets/bright.py

```diff
@@ -0,0 +1,338 @@
+    """
+
+    classes = ('background', 'intact', 'damaged', 'destroyed')
+
+    colormap = (
+        'white',  # background
+        'green',  # intact
+        'burlywood',  # damaged
+        'red',  # destroyed
+    )
+
+    md5 = '2c435bb50345d425390eff59a92134ac'
+
+    url = 'https://huggingface.co/datasets/Kullervo/BRIGHT/resolve/50901f05db4acbd141e7c96d719d8317910498fb/dfc25_track2_trainval.zip'
+
+    data_dir = 'dfc25_track2_trainval'
+
+    valid_split = ('train', 'val', 'test')
+
+    # train_setlevels.txt are the training samples
+    # holdout_setlevels.txt are the validation samples
+    # val_setlevels.txt are the test samples
+    split_files: ClassVar[dict[str, str]] = {
+        'train': 'train_setlevel.txt',
+        'val': 'holdout_setlevel.txt',
+        'test': 'val_setlevel.txt',
+    }
+
+    px_class_values: ClassVar[dict[int, str]] = {
+        0: 'background',
+        1: 'intact',
+        2: 'damaged',
+        3: 'destroyed',
+    }
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new BRIGHT DFC2025 dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train/val/test split to load
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            AssertionError: If *split* is not one of 'train', 'val', or 'test.
+        """
+        assert split in self.valid_split, f'Split must be one of {self.valid_split}'
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        self.sample_paths = self._get_paths()
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and target at that index
+        """
+        idx_paths = self.sample_paths[index]
+
+        pre_image = self._load_image(idx_paths['pre_image']).float()
+        post_image = self._load_image(idx_paths['post_image']).float()
+        # https://github.com/ChenHongruixuan/BRIGHT/blob/11b1ffafa4d30d2df2081189b56864b0de4e3ed7/dfc25_benchmark/dataset/make_data_loader.py#L101
+        # post image is stacked to also have 3 channels
+        post_image = repeat(post_image, 'c h w -> (repeat c) h w', repeat=3)
```

Comment:
Hmm, does this make sense? Is this only because they want to use RGB models pre-trained on ImageNet?

Reply:
It depends. If we want to use one of our S1 pretrained models, not duplicating may make more sense. I would personally leave it up to the user to add a transform if they want this.

Reply:
My thinking was that this is how it is implemented in the challenge code, so I thought it would make sense to follow that. I think a constant nummer of channels across the two images makes the dataset easier to use. For anything more specific, I would expect that a user has to write their own collate function/augmentations anyways. Also pretrained RGB models are tough to beat, even if trained on ImageNet https://arxiv.org/abs/2305.13456. And I don't think it would prevent you from using a pretrained S1 model, you would have to write a custom trainer anways to handle the inputs I believe.

Reply:
Also note that the current pretrained S1 models https://torchgeo.readthedocs.io/en/stable/api/models.html#sentinel-1 expect 2 input channels, so they would also not work without a channel repeat for this dataset.

Reply:
Alright, those are convincing arguments. We can consider dropping this for the non-DFC2025 version of the dataset.